### ISSUE TITLE
Is this just me with audacity?

### ISSUE THREAD

**Issue by StuartIanNaylor** [atomic14/ICS-43434-breakout-board#4]
If I import mono 1600hz 32bitPCM raw in audacity then the playback rate seems likes its doubled.

If I import mono 1600hz 16bitPCM raw in audacity then the playback is normal.

Confused as

// i2s config - this is set up to read fro the left channel
i2s_config_t i2s_config = {
    .mode = (i2s_mode_t)(I2S_MODE_MASTER | I2S_MODE_RX),
    .sample_rate = 16000,
    .bits_per_sample = I2S_BITS_PER_SAMPLE_32BIT,
    .channel_format = I2S_CHANNEL_FMT_ONLY_LEFT,
    .communication_format = i2s_comm_format_t(I2S_COMM_FORMAT_I2S),
    .intr_alloc_flags = ESP_INTR_FLAG_LEVEL1,
    .dma_buf_count = 4,
    .dma_buf_len = 1024,
    .use_apll = false,
    .tx_desc_auto_clear = false,
    .fixed_mclk = 0};

**Comment by StuartIanNaylor** [comment]
Yeah I got one mic that is duff replaced it with another and the new one works fine.
Its now a stereo pair.
https://drive.google.com/open?id=1FQkvVXz-4xSGMut8JCYTe96_uW_Absuy

I am still confused about audacity as it has to be imported as a 16bit pcm stereo 44100 as using these settings.

```
// i2s config - this is set up to read fro the left channel
i2s_config_t i2s_config = {
    .mode = (i2s_mode_t)(I2S_MODE_MASTER | I2S_MODE_RX),
    .sample_rate = 44100,
    .bits_per_sample = I2S_BITS_PER_SAMPLE_32BIT,
    .channel_format = I2S_CHANNEL_FMT_RIGHT_LEFT,
    .communication_format = i2s_comm_format_t(I2S_COMM_FORMAT_I2S | I2S_COMM_FORMAT_I2S_MSB),
    .intr_alloc_flags = ESP_INTR_FLAG_LEVEL1,
    .dma_buf_count = 4,
    .dma_buf_len = 1024,
    .use_apll = false,
    .tx_desc_auto_clear = false,
    .fixed_mclk = 0};
```

So with the above I thought I should be importing a raw stereo 32bitpcm @ 44100 but maybe someone can try and just tell me if its me being dumb as it certainly sounds like 2x 16bits becomes 1x 32 so the playback doubles in speed.
Its actually no bother as 16bit is fine but I2S_BITS_PER_SAMPLE_32BIT or I2S_BITS_PER_SAMPLE_24BIT has me confused to what is happening.

**Comment by cgreening** [comment]
In the `I2SSampler.cpp` code we convert from the 32-bit samples to 16 bit.

I2S actually only really gives you 24-bit samples.

```c++
void I2SSampler::processI2SData(uint8_t *i2sData, size_t bytes_read)
{
    int32_t *samples = (int32_t *)i2sData;
    for (int i = 0; i < bytes_read / 4; i++)
    {
        // you may need to vary the >> 11 to fit your volume - ideally we'd have some kind of AGC here
        addSample(samples[i] >> 11);
    }
}
```

**Comment by StuartIanNaylor** [comment]
Lols my C is so poor I am having to take your word for it even with looking at the code :)

But good its not me being dumb with audacity.
The reason I was trying to set 24bit is headroom and also like you say the mems are 24bit .
With my extremely limited knowledge I was just trying to work out where I could sum a stereo pair.
Its that thing again https://3cfeqx1hf82y3xcoull08ihx-wpengine.netdna-ssl.com/wp-content/uploads/2015/02/Microphone-Array-Beamforming.pdf

48khz is approx 7mm a sample but maybe a directional array could be 70mm so a lower sample rate could handle it... anyway :)

Chris have you played with any of the ADF functions as the resample would mix stereo to mono and because of the high SNR and sensitivity the gain is more than the summed SNR.
Is the resample quite heavy and that is why you used the above rather than sticking with the framework?

**Comment by StuartIanNaylor** [comment]
@cgreening I did notice the https://www.microchipdirect.com/product/ZL38063LDG1 is pretty cheap.
Order Quantity GBP per Unit
1-24 £4.12, 25-99 £3.79, 100+ £3.39.

Its the audio processor on the Espressif ESP32 LyraTD MSC used to improve the audio signal as a Wrover 2/3x I2S mic and a possible ZL38063 board?
It adds AEC for audio barge in and presume has beamforming for far field enhancement.

Thing is standalone units with audio out with micro amps and speakers are actually pretty woeful and you might want to use wifi or wired speakers instead of something of better quality.
As far as I can make out AEC needs to be on the playing codec and you might not want that 'micro' sound so AEC on the ZL38063 becomes redundant which leaves the beamforming which also suffers badly in the presence of predominant voice noise from TV or HiFi.

So hence why I got excited about the simplicity of your later 'Alexa' and also making it more simple by being a KWS activated microphone.
 I keep farting on about directionality just to help remove omidirectional noise by having nearest mic point roughly your direction as a wrover & 2x i2s mics as multiple distributed units are likely to work better by chosing the one with the best KW hit.
They are cheap enough that x2 or x4 units in a room is still pretty cheap.

This might also be of interest https://github.com/jorgenkraghjakobsen/snapclient as it uses network time to adjust latency which is pretty important with ear sensitivity. Still a WIP and hence why I have been using a Pi version https://github.com/badaix/snapcast and dac, 50watt amp and some pretty nice bookshelf speakers.
